src/hints/block_context.rs

```diff
@@ -176,6 +147,141 @@ pub fn load_deprecated_class(
     Ok(hint_extension)
 }
 
+pub const LOAD_CLASS_FACTS: &str = indoc! {r#"
+    ids.compiled_class_facts = segments.add()
+    ids.n_compiled_class_facts = len(os_input.compiled_classes)
+    vm_enter_scope({
+        'compiled_class_facts': iter(os_input.compiled_classes.items()),
+        'compiled_class_visited_pcs': os_input.compiled_class_visited_pcs,
+    })"#
+};
+pub fn load_class_facts(
+    vm: &mut VirtualMachine,
+    exec_scopes: &mut ExecutionScopes,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    let os_input = exec_scopes.get::<StarknetOsInput>("os_input")?;
+    let compiled_class_facts_ptr = vm.add_memory_segment();
+    insert_value_from_var_name("compiled_class_facts", compiled_class_facts_ptr, vm, ids_data, ap_tracking)?;
+
+    insert_value_from_var_name("n_compiled_class_facts", os_input.compiled_classes.len(), vm, ids_data, ap_tracking)?;
+
+    let compiled_class_facts: Box<dyn Any> = Box::new(os_input.compiled_classes.into_iter());
+    let compiled_class_visited_pcs: Box<dyn Any> = Box::new(os_input.compiled_class_visited_pcs);
+    exec_scopes.enter_scope(HashMap::from([
+        (String::from("compiled_class_facts"), compiled_class_facts),
+        (String::from("compiled_class_visited_pcs"), compiled_class_visited_pcs),
+    ]));
+    Ok(())
+}
+
+//
+pub const LOAD_CLASS_INNER: &str = indoc! {r#"
+    from starkware.starknet.core.os.contract_class.compiled_class_hash import (
+        create_bytecode_segment_structure,
+        get_compiled_class_struct,
+    )
+
+    compiled_class_hash, compiled_class = next(compiled_class_facts)
+
+    bytecode_segment_structure = create_bytecode_segment_structure(
+        bytecode=compiled_class.bytecode,
+        bytecode_segment_lengths=compiled_class.bytecode_segment_lengths,
+        visited_pcs=compiled_class_visited_pcs[compiled_class_hash],
+    )
+
+    cairo_contract = get_compiled_class_struct(
+        identifiers=ids._context.identifiers,
+        compiled_class=compiled_class,
+        bytecode=bytecode_segment_structure.bytecode_with_skipped_segments()
+    )
+    ids.compiled_class = segments.gen_arg(cairo_contract)"#
+};
+pub fn load_class_inner(
+    vm: &mut VirtualMachine,
+    exec_scopes: &mut ExecutionScopes,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    let class_iter = exec_scopes.get_mut_ref::<IntoIter<Felt252, CasmContractClass>>("compiled_class_facts")?;
+
+    let (compiled_class_hash, class) = class_iter.next().unwrap();
```

Comment:
Isn't this `unwrap()`  bound to panic at the end of the iterator?

Reply:
End of the iterator won't be reached it everything goes well (iteration is driven from cairo). Added more explicit error message.